Form/ChoiceList/ModelChoiceList.php

```diff
@@ -20,7 +20,18 @@
 use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 use Symfony\Component\PropertyAccess\PropertyPath;
 
+@trigger_error(
+    'The '.__NAMESPACE__.'\ModelChoiceList class is deprecated since SF 2.7'.
```

Comment:
@Soullivaneuh That's the issue. If you remove the `ModelChoiceList` class in 4.0 but still support SF < 2.7 in 4.0, then the `ModelChoiceLoader` class will not be usable by SF < 2.7 setups. Because the class `ArrayChoiceList` (which is used in the `ModelChoiceLoader` [here](https://github.com/sonata-project/SonataAdminBundle/blob/1d9ef6e/Form/ChoiceList/ModelChoiceLoader.php#L133)) exists only since SF >= 2.7.

In case Sonata 4.0 does not drop SF 2.7 support, that means I need to override the call to `ArrayChoiceList` class, to check if it exists, and fallback to `SimpleChoiceList` class if it does not.